pkg/core/app/hooks.go

```diff
@@ -0,0 +1,29 @@
+package app
+
+import (
+	"go.keploy.io/server/v2/pkg/platform/docker"
+	"go.uber.org/zap"
+)
+
+// AppRuntimeHooks defines extension points used during app lifecycle.
+// Implementations may mutate compose spec in-place.
+type AppRuntimeHooks interface {
+	BeforeDockerComposeSetup(logger *zap.Logger, compose *docker.Compose, serviceName string) (bool, error)
+	BeforeDockerSetup(logger *zap.Logger, cmd string) (string, error)
+}
+
+// RuntimeHooks is the singleton used by runtime; can be overridden by other builds.
+var RuntimeHooks AppRuntimeHooks = defaultAppRuntimeHooks{}
+
+type defaultAppRuntimeHooks struct{}
+
+func (defaultAppRuntimeHooks) BeforeDockerComposeSetup(logger *zap.Logger, _ *docker.Compose, _ string) (bool, error) {
```

Comment:
In both methods, we should not pass the logger as an argument. And context should be the first argument to be passed in both methods.